### DISCUSSION TITLE
Need Help: How do I build a distributed macOS vector database for my app

### DISCUSSION THREAD

**Discussion by product-manager2** [milvus-io/milvus#25625] [Q&A and General discussion]
Hello Everyone,

I'm currently working on a project that I believe will be of interest to many of you. I'm trying to build an open-source vector search engine, with the initial version designed for macOS.

Our initial approach involves supporting the network via hosted Milvus instances. However, as we progress, we're looking to shift towards a more user-driven model. The idea is to have our app users run nodes on their own systems, which would spring into action only when a search request is initiated within the network.

I'm reaching out to this knowledgeable community for advice on how to seamlessly integrate the Milvus database into my Swift application, and how to ensure effective communication within this distributed network. Any tips, resources, or pointers you could provide would be immensely helpful.

Thank you for taking the time to read this and for any insights you might share.

**Comment by xiaofan-luan** [comment]
> Hello Everyone,
> 
> I'm currently working on a project that I believe will be of interest to many of you. I'm trying to build an open-source vector search engine, with the initial version designed for macOS.
> 
> Our initial approach involves supporting the network via hosted Milvus instances. However, as we progress, we're looking to shift towards a more user-driven model. The idea is to have our app users run nodes on their own systems, which would spring into action only when a search request is initiated within the network.
> 
> I'm reaching out to this knowledgeable community for advice on how to seamlessly integrate the Milvus database into my Swift application, and how to ensure effective communication within this distributed network. Any tips, resources, or pointers you could provide would be immensely helpful.
> 
> Thank you for taking the time to read this and for any insights you might share.

what is the qps and latency estimation of you your request? 
load data from remote storage may take very long to support ANN

**Reply by product-manager2** [reply]
It's okay if it's slow. I want to build it for people who can't afford to pay. So they should be able to run their own clusters and host some data on their devices for querying by others as well. The hosted Milvus takes care of the sync, duplication management as well as hosting caches near edge users for more frequently demanded resources.

Think about this use case. 

A team chat app, where the team's chat messages are vectored and kept on each team member's devices. Some part of their work the team might want to be publicly available, so they can host it on both Zilliz and their own servers. Every time there's a query, whichever Mac is available returns the data depending on the proximity to data. Their public docs could also be hosted by other users in the network to improve local availability.
